power-of-2/power-of-2.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+import "fmt"
+
+func main() {
+	fmt.Println("The program determines if a number is a power of 2.")
+	var n, i int
+	fmt.Print("Enter the number: ")
+	fmt.Scan(&n)
+	if n > 0 && (n&(n-1)) == 0 {
```

Comment:
Powers of 2 contains in binary system only one 1byte, that means when we take away this one 1 with expression n&(n-1) we check if in result will be all only 0 bytes, and if they are, that means that number is power of 2. In exception number 0 already contains only 0 in its binary system, and counts as power of 2.

Reply:
This is not *exactly* correct since you mix up bits and bytes. But regardless, we are going to consider your answer valid, since you do understand what is going on there, and we do not expect you to use all of the terminology correctly yet.

You understand that other students wouldn't use the same solution, since bits operation isn't something we have learned. Please try to solve homework assignments using the tools that everyone is familiar with. However, please include extra solutions like this one - this is walking an extra mile, and we definitely will give you some extra credit for these advanced solutions.